src/App.spec.js

```diff
@@ -1,34 +1,52 @@
-import { getRoot } from './TestTools/TestCompositionRoot'
+import { initialState } from './Store/initial';
+import { makeReducers } from './Store/reducer';
 import { describe, expect, it, vi } from 'vitest';
+import StubHttpGateway from './Checkbox/StubHttpGateway';
+import { checkedAndUnchecked } from './Store/selector';
 
 describe('Checkbox App', () => {
   it('is initially unchecked', async () => {
-    const { checkboxPresenter } = getRoot();
-    expect(checkboxPresenter.isChecked).toBe(false)
-  })
-
-  it('chages from checked to unchecked and back', async () => {
-    const { checkboxPresenter } = getRoot();
-
-    checkboxPresenter.setIsChecked(true)
-    expect(checkboxPresenter.isChecked).toBe(true)
+    expect(
+      {isChecked: false},
+      makeReducers(new StubHttpGateway())(initialState, {})
+    );
+  });
 
-    checkboxPresenter.setIsChecked(false)
-    expect(checkboxPresenter.isChecked).toBe(false)
+  it('makes an external request when checked', () => {
+    const httpGateway = new StubHttpGateway();
+    httpGateway.post = vi.fn().mockImplementation(() => {
+      return Promise.resolve({});
+    });
 
-    checkboxPresenter.setIsChecked(true)
-    expect(checkboxPresenter.isChecked).toBe(true)
+    expect(
+      {isChecked: true},
+      makeReducers(httpGateway)(initialState, {type: 'SET_IS_CHECKED', payload: true})
+    );
+    expect(httpGateway.post).toHaveBeenCalledWith('is-checked', {
+      checked: true,
+    });
   })
 
-  it('makes an external request when checked', async () => {
-    const { checkboxPresenter, httpGateway } = getRoot();
+  it('does not make an external request when unchecked', () => {
+    const httpGateway = new StubHttpGateway();
     httpGateway.post = vi.fn().mockImplementation(() => {
-      return Promise.resolve({})
+      return Promise.resolve({});
     });
 
-    checkboxPresenter.setIsChecked(true)
-    expect(httpGateway.post).toHaveBeenCalledWith('is-checked', {
-      checked: true
-    })
+    expect(
+      {isChecked: true},
+      makeReducers(httpGateway)(initialState, {type: 'SET_IS_CHECKED', payload: false})
+    );
+    expect(httpGateway.post).not.toHaveBeenCalledWith('is-checked', {
+      checked: false,
```

Comment:
The unit tests are completely decoupled from the React components.

This unit test doesn't know anything about React.

And it still tests really thoroughly: that the state was set to `isChecked: true`, and this made an HTTP request.